analysis/src/python/data_collection/api/platform_objects.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Generic, List, Optional, TypeVar
+
+
+class Platform(str, Enum):
+    HYPERSKILL = 'hyperskill'
+    STEPIK = 'stepik'
+
+    @classmethod
+    def values(cls):
+        return list(map(lambda c: c.value, cls))
+
+
+@dataclass
+class Object:
+    pass
+
+
+@dataclass
+class BaseRequestParams:
+    page: int = 1
+    page_size: int = 100
+    ids: Optional[List[int]] = None
+
+
+@dataclass
+class Meta:
+    page: int
+    has_next: bool
+    has_previous: bool
+
+
+T = TypeVar('T', bound='Object')
+
+
+@dataclass
+class ObjectResponse(Generic[T]):
+    meta: Meta
+
+    def get_objects(self) -> List[T]:
+        pass
```

Comment:
Why is it ok?

Reply:
It is not beautiful, I know, but the reason was following

Reply:
then you can raise not implemented error here)